pytest_mpi/plugin.py

```diff
@@ -304,44 +303,34 @@ class MPIImplementation(enum.Enum):
 
 
 def detect_mpi_implementation() -> MPIImplementation:
```

Comment:
While we're doing this could you wrap this in `@functools.cache` since we only want to call this once?